File: Unity/Messages.cs
namespace BlenderBridge
{
    public enum ServerMessageType : byte
    {
        Heartbeat = 0,
        DomainReload = 1
    }

    public enum ClientMessageType : byte
    {
        ViewUpdated = 0,
        ViewDestroyed = 1,
        ObjectVisibility = 2,
    }

    struct DomainReload : Comms.IMessage
    {
        public byte Type => (byte)ServerMessageType.DomainReload;
    }

    struct Heartbeat : Comms.IMessage
    {
        public byte Type => (byte)ServerMessageType.Heartbeat;
    }
}

File: Blender~/messages.py
from enum import IntEnum, unique
from .comms.message import IMessage


@unique
class ServerMessageType(IntEnum):

    Heartbeat = 0
    DomainReload = 1


@unique
class ClientMessageType(IntEnum):

    ViewUpdated = 0
    ViewDestroyed = 1
    ObjectVisibility = 2


class Heartbeat(IMessage):

    @staticmethod
    def get_type():
        return ClientMessageType.Heartbeat

    def __init__(self):
        pass

    def initialize(self):
        return self

    def serialize(self):
        data = self.__dict__
        return data

    def post_deserialize(self):
        pass


class DomainReload(IMessage):

    @staticmethod
    def get_type():
        return ServerMessageType.DomainReload

    def __init__(self):
        pass

    def initialize(self):
        return self

    def serialize(self):
        data = self.__dict__
        return data

    def post_deserialize(self):
        pass

